chevah/compat
# [#4387] Use coverator for coverage publishing.

Scope
=====

Use a new tool instead of codecov to publish coverage data.


Changes
=====

I've added the `[paths]` section to the `.coveragerc` configuration file. This is needed for better path handling when combining different coverage data files on a different machine/path where the data was gathered.

Also implemented a new publish_coverage that uses the chevah-coverage tool instead of codecov to submit the coverage data.

The new publish_coverage uses some git info that are not available in the GitBrink API so there is some low level git usage, but this should be gone when we move this to brink.


How to try and test the changes
===============================

reviewers: @adiroiban 

Please check if the changes make sense.

Running `paver test_coverage && paver coverage_publish` should publish the data to `http://coverage.chevah.com:8080`.

The server will only generate the HTML reports after 7 different builders publish reports, to simulate this manually you can run:

`BUILDER_NAME=some-name paver coverage_publish`

I'm running a test instance of the coverage server in `http://172.20.245.1:8080` so for the command above to work you need to modify your your hosts file.
